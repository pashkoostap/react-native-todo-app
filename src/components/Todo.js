import React from "react";
import { View, Text, TouchableOpacity } from "react-native";

const Todo = ({ todo, deleteTodo }) => (
  <View>
    <Text>{todo.title}</Text>

    <TouchableOpacity onPress={() => deleteTodo(todo.id)}>
      <Text>Delete todo</Text>
    </TouchableOpacity>
  </View>
);

export default Todo;
