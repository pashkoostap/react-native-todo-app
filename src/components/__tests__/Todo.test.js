import React from "react";
import renderer from "react-test-renderer";
import Todo from "../Todo";

describe("Todo", () => {
  it("should match the snapshot", () => {
    const todo = { title: "todo" };
    const rendered = renderer.create(<Todo todo={todo} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
