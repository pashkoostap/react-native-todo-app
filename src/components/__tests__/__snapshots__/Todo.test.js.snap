// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match the snapshot 1`] = `
<View>
  <View
    collapsable={undefined}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flex": 1,
        "left": 0,
        "padding": 20,
        "top": 0,
        "width": "100%",
        "zIndex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      todo
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "width": 70,
        "zIndex": 0,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "bottom": 0,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 70,
        "zIndex": 0,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
